rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java

```diff
@@ -85,6 +87,12 @@ private void  prePopulateSampleCollectionFormFields(JSONObject field, String uni
         if (CovidConstants.FormFields.PATIENT_INFO_DOB.equals(field.getString(KEY))) {
             fillPatientData(field, "Patient Dob: ", patient.getDob());
         }
+
+        // pre-populate the sampler name
+        if (CovidConstants.FormFields.SAMPLER_NAME.equals(field.getString(KEY))) {
+            final AllSharedPreferences allSharedPreference = RDTApplication.getInstance().getContext().allSharedPreferences();
+            field.put(JsonFormConstants.VALUE, allSharedPreference.getANMPreferredName(allSharedPreference.fetchRegisteredANM()));
```

Comment:
I would move this into a private method to be shared by both the sample collection and sample shipment form.

Reply:
I also think we should use `JsonFormConstants.TEXT` here, verify that it works with `JsonFormConstants.VALUE` and if so you can leave it as is.

Reply:
fixed.